v2/azure-arm.yaml

```diff
@@ -1197,6 +1209,78 @@ objectModelConfiguration:
       Profiles_Endpoint:
         $exportAs: ProfilesEndpoint
         $supportedFrom: v2.0.0-beta.1
+    2023-05-01:
+      # DeepCreatedOriginGroup actually is a resource, for some CDN SKUs it's required to be specified on the CDN endpoint
+      # while for other SKUs it seems to be optional on the endpoint (or may not be allowed to be set on the endpoint at all, I am not sure).
+      # Given the requirement at least for classic CDNs (Sku == Standard_Microsoft) to specify Origin at the Endpoint scope
+      # we can't treat this purely as a resource. Instead, I think this field is like AKS AgentPools - both embedded and not, and
+      # the author of the ASO resource must choose which makes sense for their context.
+      # This comment applies to all the "DeepCreated*" resources.
+      DeepCreatedOriginGroup:
+        $isResource: false
+      DeepCreatedOriginGroup_STATUS:
+        $isResource: false
+      DeepCreatedOriginGroupProperties:
+        $isResource: false
+      DeepCreatedOriginGroupProperties_STATUS:
+        $isResource: false
+      DeepCreatedOrigin:
+        $isResource: false
+      DeepCreatedOrigin_STATUS:
+        $isResource: false
+      DeepCreatedOriginProperties_STATUS:
+        $isResource: false
+      Profile:
+        $export: true
+        $supportedFrom: v2.6.0
+# The below resources could be exported but are part of "CDN classic".
+# This part of the CDN/AzureFrontdoor API is confusing but our understanding is that
+# the "classic" CDN is not preferred compared to AFD, so for now we aren't exporting these.
+#      Profiles_Endpoint:
+#        $exportAs: ProfilesEndpoint
+#        $supportedFrom: v2.6.0
+#      Profiles_Endpoints_OriginGroup:
+#        $exportAs: OriginGroup
+#        $supportedFrom: v2.6.0
+#      Profiles_Endpoints_Origin:
+#        $exportAs: Origin
+#        $supportedFrom: v2.6.0
+#      Profiles_Endpoints_CustomDomain:
+#        $exportAs: CustomDomain
+#        $supportedFrom: v2.6.0
+      Profiles_CustomDomain:
+        $exportAs: AfdCustomDomain
+        $supportedFrom: v2.6.0
+      Profiles_AfdEndpoint:
```

Comment:
There is `Endpoint/Origin` and `Endpoint/OriginGroup` resources as well. Do we want them too? 
https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/profiles/endpoints/origins?pivots=deployment-language-arm-template

Reply:
See comment in azure-arm.yaml for this:
```
      # DeepCreatedOriginGroup actually is a resource, for some CDN SKUs it's required to be specified on the CDN endpoint
      # while for other SKUs it seems to be optional on the endpoint (or may not be allowed to be set on the endpoint at all, I am not sure).
      # Given the requirement at least for classic CDNs (Sku == Standard_Microsoft) to specify Origin at the Endpoint scope
      # we can't treat this purely as a resource. Instead, I think this field is like AKS AgentPools - both embedded and not, and
      # the author of the ASO resource must choose which makes sense for their context.
      # This comment applies to all the "DeepCreated*" resources.
      DeepCreatedOriginGroup:
        $isResource: false
      DeepCreatedOriginGroup_STATUS:
        $isResource: false
      DeepCreatedOriginGroupProperties:
        $isResource: false
      DeepCreatedOriginGroupProperties_STATUS:
        $isResource: false
      DeepCreatedOrigin:
        $isResource: false
      DeepCreatedOrigin_STATUS:
        $isResource: false
```

It's not clear to me when they allow management via the child resource and when they don't. So I started with AFD which at least was clear to me. We can add these resources in as actual resources in the future as well but it may require merging with the `DeepCreated` variants on the parent objects - needs more testing.